### ISSUE TITLE
Consider raising number of artefacts kept by build pipelines

### ISSUE THREAD

**Issue by sxa** [AdoptOpenJDK/ci-jenkins-pipelines#15]
Current process is to disable nightly testing, but not buids, during a release cycle. For at least some pipelines, only one artefact is being kept. This means that it is highly likely that if any re-runs of test jobs are required which reference the upstream job, the artefacts will already have been removed from the individual jobs when a re-run is required, making it more complex to run the tests. We should evaluate the space impacts of increasing this to 2 builds everywhere instead of 1 for `Max # of builds to keep with artifacts`

**Comment by smlambert** [comment]
+1 times 1000 on this request... especially valuable during release periods to be able to rerun stuff (before its published).

**Comment by sxa** [comment]
@smlambert Of course the obvious concern with this is that with [HotSpot debug images being added soon](https://github.com/AdoptOpenJDK/openjdk-build/pull/2443) it will be even more of an increase to our space requirements :-)

**Comment by smlambert** [comment]
Indeed, perhaps its configurable to default to keep 1, but during release builds keep 2 until we are through the week...

**Comment by andrew-m-leonard** [comment]
Yes, I hit this issue during the last release once upstream logs dissappear, although it wasn't too difficult to update the reference to the pipeline artifact
